pkg/build/controller/strategy/util.go

```diff
@@ -649,3 +653,105 @@ func setupBlobCache(pod *corev1.Pod) {
 		pod.Spec.InitContainers = initContainers
 	}
 }
+
+// injectBuildVolumesIntoPod injects the provided BuildVolumes into given Pod
+func addBuildVolumesToPod(pod *corev1.Pod, buildVolumes []buildv1.BuildVolume) error {
+	if len(buildVolumes) == 0 {
+		return nil
+	}
+
+	for _, bv := range buildVolumes {
+		v := corev1.Volume{
+			Name: bv.Name,
+		}
+
+		switch {
+		case bv.Type == buildv1.BuildVolumeSourceTypeSecret:
+			v.Secret = bv.Secret
+		case bv.Type == buildv1.BuildVolumeSourceTypeConfigMap:
+			v.ConfigMap = bv.ConfigMap
+		default:
+			return fmt.Errorf("encountered unsupported build volume source type %q", bv.Type)
+		}
+
+		pod.Spec.Volumes = append(pod.Spec.Volumes, v)
+	}
+
+	return nil
+}
+
+// injectBuildVolumeMountsIntoContainer injects the provided BuildVolumeMounts into the given Container
+func addBuildVolumeMountsToContainer(container *corev1.Container, buildVolumeMounts []buildv1.BuildVolumeMount) error {
+	if len(buildVolumeMounts) == 0 {
+		return nil
+	}
+
+	for _, bvm := range buildVolumeMounts {
+		vm := corev1.VolumeMount{
+			Name:      bvm.Name,
+			ReadOnly:  bvm.ReadOnly,
+			MountPath: path.Join(BuildVolumeMountPath, bvm.MountPath),
```

Comment:
`bvm.MountPath` is an absolute path that we want the Secret or ConfigMap to show up in the _buildah container_. For the runners in the build pod, we should use the volume name as the mount path.

Example
Volume Name - "etc-pki-entitlement"
Mount path in build runner: "/var/run/openshift.io/volumes/etc-pki-entitlement"
Final destination in buildah build: /etc/pki/entitlement

```suggestion
			MountPath: path.Join(BuildVolumeMountPath, bvm.Name),
```

Reply:
updated in next push